app/src/components/Layers/CanvasLayer.jsx

```diff
@@ -131,6 +143,39 @@ class CanvasLayer {
     return canvas;
   }
 
+  getVesselAtPixelLocation(x, y) {
+    this.playbackData.forEach(canvasPlaybackData => {
+      const canvas = canvasPlaybackData.canvas;
+      const rect = canvas.getBoundingClientRect();
+      if (x > rect.left && y > rect.top && x <= rect.right && y <= rect.bottom) {
+        return this._getCanvasVesselAt(canvas, x - rect.left, y - rect.top);
+      }
+    });
+  }
+
+  _getCanvasVesselAt(canvas, x, y) {
```

Comment:
Yeah sorry, I was in the middle of this. I'll remove the last commit from the PR, merge, and open another WIP one.